src/commands/git.rs

```diff
@@ -18,6 +19,16 @@ pub fn add(app: &mut Application) -> Result {
     index.write().chain_err(|| "Failed to write index.")
 }
 
+fn get_gh_path(url: &str) -> errors::Result<&str> {
```

Comment:
As a follow-on to the suggested import, we can drop the `errors::` prefix here and just use `Result` directly. :sunglasses:

Reply:
The import of ```use crate::commands::{self, Result};``` 
which is ```pub type Result = errors::Result<()>;```
will shadow `Result`, I guess I have to use the mod prefix for ```errors::Result<&str>``` to work around this

Reply:
Shadowing is the intent. We want to use `errors::Result` for any `Result` type declarations in function signatures. By removing the `use crate::errors` and removing the `errors::` prefix on this function signature, you'll be using an `errors::Result` return type because of the shadowing.

This can seem a little weird if you're not used to the error handling pattern encouraged by the `error-chain` crate, but it's the pattern used throughout amp. :slightly_smiling_face:

Reply:
What is needed here is `errors::Result<&str>`, not the type aliased `errors::Result<()>`.  I'm afraid the problem here is `crate::commands::Result` is shadowing other use-cases of ```errors::Result```, which is not related to `error-chain`.

And `By removing the use crate::errors and removing the errors:: prefix on this function signature`, we will get this:

```
error[E0107]: wrong number of type arguments: expected 0, found 1
  --> src/commands/git.rs:22:37
   |
22 | fn get_gh_path(url: &str) -> Result<&str> {
   |                                     ^^^^ unexpected type argument
```

Am I misunderstanding something?

Reply:
You're right; I forgot that we're in a command module and we've already aliased `Result` as `errors::Result<()>`. Apologies for the confusion! :blush:

Reply:
😄